source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,257 @@
+        corrected_blue_balance = camera.settings.blue_balance * mean_green / mean_blue
+        with camera.update_settings() as updater:
+            updater.settings.red_balance = corrected_red_balance
+            updater.settings.blue_balance = corrected_blue_balance
+        rgb = _rgb_from_3d_capture(camera)
+        [mean_red, mean_green, mean_blue] = _compute_mean_rgb(rgb, 100)
+        _display_rgb(rgb, "RGB image after color balance (3D)")
+
+    return (
+        corrected_red_balance / default_red_balance,
+        corrected_blue_balance / default_blue_balance,
+    )
+
+
+def _display_color_balance_example_results():
+    """Load and display example images before and after color balance.
+
+    Args None
+
+    Returns None
+
+    """
+    print(f"Displaying color balance calibration example results")
+    dir_path = Path(__file__).parent / "color_balance_files"
+    _display_rgb(
+        plt.imread(str(dir_path / "white_surface_before.png")),
+        "White surface before color balance",
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "white_surface_after.png")),
+        "White surface after color balance",
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "scene_before.png")), "Scene before color balance"
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "scene_after.png")), "Scene after color balance"
+    )
+
+
+def _set_settings_3d(camera, iris, exposure_time, brightness, gain):
+    """Set settings for 3D capture.
+
+    Args:
+        camera: Zivid camera
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns None
+
+    """
+    with camera.update_settings() as updater:
+        updater.settings.iris = iris
+        updater.settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        updater.settings.brightness = brightness
+        updater.settings.gain = gain
+
+
+def _set_settings_2d(application, iris, exposure_time, brightness, gain):
+    """Set settings for 2D capture.
+
+    Args:
+        application: Zivid application
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns:
+        settings_2d: 2D capture settings
+
+    """
+    settings_2d = application.Settings2D()
+    settings_2d.iris = iris
+    settings_2d.exposure_time = datetime.timedelta(microseconds=exposure_time)
+    settings_2d.brightness = brightness
+    settings_2d.gain = gain
+
+    return settings_2d
+
+
+def _apply_color_balance(image, red_balance, blue_balance):
+    """Apply color balance to image.
+
+    Args:
+        image: Input image
+        red_balance: Red balance
+        blue_balance: Blue balance
+
+    Returns:
+        corrected_image: Image after color balance
+
+    """
+
+    corrected_image = image
+    image[:, :, 0] = corrected_image[:, :, 0] * np.float32(red_balance)
```

Comment:
Be carefult. uint8 * float32 = float32. Make sure this is the behavior you want

Reply:
I will investigate.

Reply:
`corrected_rgb = np.copy(rgb)
    corrected_rgb[:, :, 0] = rgb[:, :, 0] * red_balance / default_red_balance
    corrected_rgb[:, :, 2] = rgb[:, :, 2] * blue_balance / default_blue_balance`

Not changed. Since, corrected_rgb is first created as uint8, the multiplication will return float but will be converted to uint8 as it has to adapt to type of corrected_rgb